source/blender/draw/engines/eevee/eevee_effect_antialiasing.c

```diff
@@ -0,0 +1,369 @@
+    return;
+  }
+  else {
+    const float *viewport_size = DRW_viewport_size_get();
+    const DRWView *default_view = DRW_view_default_get();
+    float *transform_offset;
+
+    transform_offset = e_data.jitter_32[min_ii(vedata->stl->effects->taa_current_sample, 32)];
+
+    /* construct new matrices from transform delta */
+    float winmat[4][4], viewmat[4][4], persmat[4][4];
+    DRW_view_winmat_get(default_view, winmat, false);
+    DRW_view_viewmat_get(default_view, viewmat, false);
+    DRW_view_persmat_get(default_view, persmat, false);
+
+    window_translate_m4(winmat,
+                        persmat,
+                        transform_offset[0] / viewport_size[0],
+                        transform_offset[1] / viewport_size[1]);
+
+    if (g_data->view) {
+      /* When rendering just update the view. This avoids recomputing the culling. */
+      DRW_view_update_sub(g_data->view, viewmat, winmat);
+    }
+    else {
+      /* TAA is not making a big change to the matrices.
+       * Reuse the main view culling by creating a sub-view. */
+      g_data->view = DRW_view_create_sub(default_view, viewmat, winmat);
+    }
+    DRW_view_set_active(g_data->view);
+    return;
+  }
+}
+
+void EEVEE_antialiasing_draw_pass(EEVEE_Data *vedata)
+{
+  EEVEE_PrivateData *g_data = vedata->stl->g_data;
+  EEVEE_FramebufferList *fbl = vedata->fbl;
+  EEVEE_PassList *psl = vedata->psl;
+  DefaultFramebufferList *dfbl = DRW_viewport_framebuffer_list_get();
+
+  EEVEE_EffectsInfo *effects = vedata->stl->effects;
+  if (!(effects->enabled_effects & EFFECT_SMAA)) {
+    return;
+  }
+
+  // In the case we remove the taa_total_sample set to 0 at ge start
+  if (vedata->stl->effects->taa_total_sample == 1) {
+    /* AA disabled. */
+    return;
+  }
+
+  eevee_antialiasing_setup(vedata);
+
+  /**
+   * We always do SMAA on top of TAA accumulation, unless the number of samples of TAA is already
+   * high. This ensure a smoother transition.
+   * If TAA accumulation is finished, we only blit the result.
+   */
+  if (vedata->stl->effects->taa_current_sample < 16) {
+    /* In playback mode, we are sure the next redraw will not use the same viewmatrix.
+     * In this case no need to save the depth buffer. */
+    eGPUFrameBufferBits bits = vedata->stl->effects->taa_current_sample == 1 ? GPU_COLOR_BIT :
+                                                                               GPU_COLOR_BIT | GPU_DEPTH_BIT;
+    GPU_framebuffer_blit(dfbl->default_fb, 0, fbl->antialiasing_fb, 0, bits);
+  }
+  else {
+    /* Accumulate result to the TAA buffer. */
+    GPU_framebuffer_bind(fbl->antialiasing_fb);
+    DRW_draw_pass(psl->aa_accum_ps);
+    /* Copy back the saved depth buffer for correct overlays. */
+    GPU_framebuffer_blit(fbl->antialiasing_fb, 0, dfbl->default_fb, 0, GPU_DEPTH_BIT);
+  }
+
+  /* After a certain point SMAA is no longer necessary. */
+  if (vedata->stl->effects->taa_current_sample < 16) {
```

Comment:
This ´´´if (vedata->stl->effects->taa_current_sample < 16)´´´ check is similar to check from line 337. Could they unified in the check above? (and i think that the comment should put just above smaa_mix_factor definition not above ´´´if (vedata->stl->effects->taa_current_sample < 16)´´´ check)

Reply:
Other than this it seems Ok. I'm going to test it. It is exciting!!!! ,😁👍

Reply:
hello! thanks for review, i just pushed 2 or 3 more cleanup commits. Thanks very much for review and happy testing. I hope this won't crash : )